react/week3/week3-react/src/components/TodoItems/TodoItems.js

```diff
@@ -0,0 +1,94 @@
+import React from "react";
+import "./TodoItems.css";
+
+const TodoItems = (props) => {
+  const { items } = props;
+  const { deletedItem, completedItem, editItem, handleChange } = props;
+  let length = items.length;
+  const ListItems = length ? (
+    items.map((item) => {
+      return (
+        <div className="ListItems" key={item.id}>
+          <span
+            className={
+              item.completed //checking whether todo item completed then marking it completed by striking line
+                ? "description list-item completed"
+                : "description list-item"
+            }
+          >
+            {item.editing ? ( //check if todo item enabled for editing if true display input box for updating text otherwise display text description
+              <input
+                type="text"
+                placeholder="Enter Todo description"
+                id="description"
+                onChange={(e) => handleChange(item.id, e)}
+                value={item.description}
+                required
+              ></input>
+            ) : (
+              item.description
+            )}
+          </span>
+          <span
+            className={
+              item.completed ? "when list-item completed" : "when list-item"
+            }
+          >
+            {item.editing ? ( //check if todo item enabled for editing if true display input box for updating text otherwise display text description
+              <input
+                type="text"
+                placeholder="Enter time and date"
+                id="when"
+                onChange={(e) => handleChange(item.id, e)}
+                value={item.when}
+                required
+              ></input>
+            ) : (
+              item.when
+            )}
+
+
+          </span>
+          <span class="action icon">
+            <input
+              type="checkbox"
+              checked={item.completed ? "checked" : ""}
```

Comment:
```suggestion
              defaultChecked={item.completed ? "checked" : ""}
```
You provided a `checked` prop to a form field without an `onChange` handler. This will render a read-only field. If the field should be mutable use `defaultChecked`